here-naksha-app-service/src/test/java/com/here/naksha/app/auth/JwtTest.java

```diff
@@ -0,0 +1,121 @@
+package com.here.naksha.app.auth;
+
+
+import com.here.naksha.app.common.ApiTest;
+import com.here.naksha.app.service.http.auth.NakshaAuthProvider;
+import com.here.naksha.lib.core.util.IoHelp;
+import com.here.naksha.lib.core.util.IoHelp.LoadedBytes;
+import com.here.naksha.lib.hub.NakshaHubConfig;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.JWTOptions;
+import io.vertx.ext.auth.PubSecKeyOptions;
+import io.vertx.ext.auth.jwt.JWTAuthOptions;
+import org.junit.jupiter.api.Test;
+
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import static com.here.naksha.app.common.assertions.ResponseAssertions.assertThat;
+
+public class JwtTest extends ApiTest {
+    // For this test suite, the default test-config.json denotes that
+    // the service is launched in DUMMY auth mode
+    @Test
+    public void testDummyModeNoJWT() throws Exception {
+        final String streamId = UUID.randomUUID().toString();
+        // Providing no JWT, the master token should be employed automatically
+        HttpResponse<String> response = getNakshaClient().get("hub/storages", streamId);
+        assertThat(response).hasStatus(200);
+    }
+
+    @Test
+    public void testDummyModeInvalidJWT() throws Exception {
+        final String streamId = UUID.randomUUID().toString();
+        // Providing an invalid JWT, should return HTTP 401 code
+        HttpResponse<String> response = getNakshaClient().get("hub/storages", streamId, "Bearer rdzftugzhkjn");
+        assertThat(response).hasStatus(401);
+    }
+
+    @Test
+    public void testDummyModeJWTSignedByUnknownKey() throws Exception {
+        final String streamId = UUID.randomUUID().toString();
+        // Providing an invalid JWT, should return HTTP 401 code
+        HttpResponse<String> response = getNakshaClient().get("hub/storages", streamId, "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c");
```

Comment:
This token uses HS256 algorithm. Which is good test to have. But I would recommend atleast one test where the JWT uses RS256 algorithm, BUT uses some different private key to sign the JWT (so basically intention is to test the signature mismatch, than the algorithm mismatch). And you can generate this JWT with some different private key which is part of Test resources folder, so it doesn't get accidentally used in real code.